EpiAware/src/EpiLatentModels/CombineLatentModels.jl

```diff
@@ -0,0 +1,62 @@
+@doc raw"
+The `CombineLatentModels` struct.
+
+This struct is used to combine multiple latent models into a single latent model.
+
+# Constructors
+
+- `CombineLatentModels(models::M) where {M <: AbstractVector{<:AbstractTuringLatentModel}}`: Constructs a `CombineLatentModels` instance with specified models, ensuring that there are at least two models.
+- `CombineLatentModels(; models::M) where {M <: AbstractVector{<:AbstractTuringLatentModel}}`: Constructs a `CombineLatentModels` instance with specified models, ensuring that there are at least two models.
+
+# Examples
+
+```julia
+using EpiAware, Distributions
+combined_model = CombineLatentModels([Intercept(Normal(2, 0.2)), AR()])
+latent_model = generate_latent(combined_model, 10)
+latent_model()
+```
+"
+@kwdef struct CombineLatentModels{M <: AbstractVector{<:AbstractTuringLatentModel}} <:
+              AbstractTuringLatentModel
+    "A vector of latent models"
+    models::M
+
+    function CombineLatentModels(models::M) where {M <:
+                                                   AbstractVector{<:AbstractTuringLatentModel}}
+        @assert length(models)>1 "At least two models are required"
+        return new{AbstractVector{<:AbstractTuringLatentModel}}(models)
+    end
+end
+
+@doc raw"
+Generate latent variables using a combination of multiple latent models.
+
+# Arguments
+- `latent_models::CombineLatentModels`: An instance of the `CombineLatentModels` type representing the collection of latent models.
+- `n`: The number of latent variables to generate.
+
+# Returns
+- `combined_latents`: The combined latent variables generated from all the models.
+- `latent_aux`: A tuple containing the auxiliary latent variables generated from each individual model.
+
+# Example
+"
+@model function EpiAwareBase.generate_latent(latent_models::CombineLatentModels, n)
+    @submodel final_latent, latent_aux = _accumulate_latents(
+        latent_models.models, 1, fill(0.0, n), [], n, length(latent_models.models))
```

Comment:
Have you tested this for AD/would you like me to? 

```julia
fill(0.0, n)
```
Would default to a Vector{Float64} which might be a problem for using `DualNumbers`.

Reply:
I haven't that is a good point. Can we instead just define it differently?

Reply:
Usually, I go with the `eltype` of a generated variable from `~` e.g. like [here](https://github.com/CDCgov/Rt-without-renewal/blob/4ab6cc2e8ac4a0e89f0973324544e0b1953a195b/EpiAware/src/EpiLatentModels/RandomWalk.jl#L91C5-L91C39). I'll have a think about how to do this case.